src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,710 @@
+            file is saved. If `False`, download is temporarily saved
+            and then merged to master and deleted.
+        """
+
+        url = (
+            f"https://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.FEWS/"
+            f".Africa/.DAILY/.ARC2/.daily/.est_prcp/T/"
+            f"%28{date_min_dl.day}%20{date_min_dl:%b%%20%Y}%29"
+            f"%28{date_max_dl.day}%20{date_max_dl:%b%%20%Y}%29RANGEEDGES/"
+            f"X/%28{self.range_x[0]}%29%28{self.range_x[1]}%29RANGEEDGES/"
+            f"Y/%28{self.range_y[0]}%29%28{self.range_y[1]}%29RANGEEDGES/"
+            f"data.nc"
+        )
+
+        raw_filepath = self._get_raw_filepath(master=master)
+
+        if os.path.exists(raw_filepath) and master:
+            logger.info(
+                "Removing existing master raw ARC2 file before re-downloading."
+            )
+            os.remove(raw_filepath)
+
+        cookies = {
+            "__dlauth_id": os.getenv("IRI_AUTH"),
+        }
+
+        logger.info(
+            f"Downloading ARC2 data from {self.date_min:%d %b %Y} to "
+            f"{self.date_max:%d %b %Y} "
+            f"covering longitudes {self.range_x[0]} to {self.range_x[1]} "
+            f"and latitudes {self.range_y[0]} to {self.range_y[1]}."
+        )
+
+        # TODO: explore specific errors generated by request
+        try:
+            response = requests.get(url, cookies=cookies, verify=False)
+        except requests.exceptions.RequestException as e:
+            raise SystemExit(e)
+
+        if response.status_code == 404:
+            logger.error("No data available from %s.", url)
+            return
+
+        # create folders if necessary and write file
+        Path(raw_filepath.parent).mkdir(parents=True, exist_ok=True)
+
+        with open(raw_filepath, "wb") as fd:
+            for chunk in response.iter_content(chunk_size=128):
+                fd.write(chunk)
+
+        # merge to master and delete temporary files
+        if not master:
+            master_filepath = self._get_raw_filepath(master=True)
+            if not os.path.isfile(master_filepath):
+                logger.error(
+                    (
+                        "Master raster data file not set. This needs "
+                        "to be done once by running "
+                        "`download_data(master=True)` "
+                        "for a unique set of coordinates."
+                    )
+                )
+
+            logger.info(
+                f"Merging ARC2 data from {self.date_min:%d %b %Y} "
+                f"to {self.date_max:%d %b %Y} "
+                f"into master file: {os.path.basename(master_filepath)}."
+            )
+
+            with xr.open_dataarray(master_filepath) as ds:
+                master = ds.load()
+            with xr.open_dataarray(raw_filepath) as ds:
+                raw = ds.load()
+
+            master_merge = xr.concat([master, raw], dim="T")
+
+            os.remove(raw_filepath)
```

Comment:
This is something I learnt from Monica recently. You always prefer not to remove files as it increases the risk that you accidentally remove something that you don't want to remove. A solution is to instead use a temporary file/folder. I am not too sure what the best structure would be here, but what I did for FN is [here](https://github.com/OCHA-DAP/pa-aa-toolbox/blob/9390ea80a8d2dc5348e04611f5cb4ba45b0fc2f6/src/aatoolbox/datasources/fewsnet/fewsnet.py#L55). Monica might have better advice

Reply:
Good idea. have added as TODO for transition to `pa-aa-toolbox`.